libraries/rush-lib/src/api/PnpmLockValidationConfiguration.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { AlreadyReportedError, FileSystem, JsonFile, JsonSchema } from '@rushstack/node-core-library';
+import fs from 'fs';
+import yaml from 'js-yaml';
+import type { Lockfile } from '@pnpm/lockfile-types';
+import type { ITerminal } from '@rushstack/terminal';
+
+import schemaJson from '../schemas/pnpm-lock-validation.schema.json';
+import type { RushConfiguration } from './RushConfiguration';
+import { CustomTipId } from './CustomTipsConfiguration';
+
+export interface IPnpmValidationJson {
+  pnpmLockValidationRules?: Array<{
+    ruleName: 'validateIntegritySha1';
```

Comment:
Consider a rule name that better indicates the goal, something like `requireSecureShaHashes` or `disallowInsecureSha1`